src/pipeline/__tests__/pipeline.test.ts

```diff
@@ -248,4 +328,86 @@ describe('Pipeline', () => {
     assertEquals(plannedPipeline.steps.length, 1);
     assertEquals(plannedPipeline.steps[0].action, 'update');
   });
+
+  it('pipeline edges are flipped when removing nodes and previous pipeline created/updated them', async () => {
```

Comment:
I think I understand what this test is doing, but this description is really bad. Can you make it something like:

```
it('should reverse edges when removing nodes', () => { ... })
```

Reply:
yeah this is a much better description 😂